build/xtask/src/flash.rs

```diff
@@ -11,59 +12,233 @@ use anyhow::Context;
 
 use crate::Config;
 
-pub fn run(verbose: bool, cfg: &Path) -> anyhow::Result<()> {
-    ctrlc::set_handler(|| {}).expect("Error setting Ctrl-C handler");
+//
+// We allow for enough information to be put in the archive for the image to
+// be flashed based only on the archive (e.g., by Humility).  Because flashing
+// is itself a bit of a mess (requiring different programs for different
+// targets), this is a bit gritty (e.g., any required external configuration
+// files must themselves put in the archive).  If these structures need to
+// change, be sure to make corresponding changes to Humility.
+//
+#[derive(Debug, Serialize)]
```

Comment:
One day, we need a common crate so Hubris and Humility can share stuff like this (but that shouldn't block this PR).

Reply:
Agreed and agreed.